### PR TITLE
Improve CMake support. Add CMake find module.

### PR DESCRIPTION
**Pull request by Ghabry**
Should work.

CMakeLists.txt is now more modern, added a file list because this is better style than globbing.

Replaced the Find modules with the official ones from latest CMake. ~~EXPAT is already shipped with CMake <= 3.7, not needed anymore.~~ EXPAT with exported targets is newer.

The -config.cmake file uses some creativity to allow a generation through autoconf.

### CODE REVIEW

**Comment by Ghabry** [comment]
@fdelapena I hope you didn't approve this without testing if the CMake build of Player works and links for you :P

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,91 +1,195 @@
...
+# installation
+include(GNUInstallDirs)
+
+install(
+	TARGETS lcf
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+# installation
```
Comment (carstene1ns) [inline CMakeLists.txt]: duplicated, unneeded comment

File: builds/liblcf-config.cmake.in [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+if(LIBLCF_FOUND)
+	set(LIBLCF_LIBRARIES ${LIBLCF_LIBRARY})
+	set(LIBLCF_LINK_LIBRARIES "")
+	set(LIBLCF_INCLUDE_DIRS ${LIBLCF_INCLUDE_DIR})
+	
+	if(NOT TARGET liblcf::liblcf)
+		add_library(liblcf::liblcf UNKNOWN IMPORTED)
+		
+		set(LIBLCF_LIBRARY_DEPS "icu-i18n expat")
+		string(REPLACE " " ";" LIBLCF_LIBRARY_DEPS "${LIBLCF_LIBRARY_DEPS}")
```
Comment (carstene1ns) [inline builds/liblcf-config.cmake.in]: why not making this a list in the first place?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,91 +1,195 @@
...
+file(GLOB includefiles "${CMAKE_CURRENT_SOURCE_DIR}/src/*.h"
+	"${CMAKE_CURRENT_SOURCE_DIR}/src/generated/*.h")
+
+install(
+	FILES ${includefiles}
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/liblcf)
+	
+# pkg-config file generation
```
Comment (carstene1ns) [inline CMakeLists.txt]: duplicated, unneeded comment

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,91 +1,195 @@
...
-include(CheckCXXCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-if(COMPILER_SUPPORTS_CXX11)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+# icu
+if(DISABLE_ICU)
+	find_package(Iconv REQUIRED)
+	target_link_libraries(lcf Iconv::Iconv)
+	if(NOT Iconv_IS_BUILT_IN)
+		list(APPEND LIBLCF_DEPS "iconv")
```
Comment (carstene1ns) [inline CMakeLists.txt]: not sure this does the right thing. `requires` and `requires.private` are for dependencies that `pkg-config` is able to resolve. however, gnu libiconv does not ship a pkg-config file (AFAIR) and a system library used on some unknown platform likely neither.

File: builds/liblcf-config.cmake.in (hunk 2 above)
Comment (Ghabry) [reply builds/liblcf-config.cmake.in]: Oh thats a bad edit, the string is supposed to be ``set(LIBLCF_LIBRARY_DEPS "@AX_PACKAGE_REQUIRES_PRIVATE@")``

**Comment by carstene1ns** [comment]
> EXPAT is already shipped with CMake <= 3.7, not needed anymore.
And yet, you included it :P

**Comment by Ghabry** [comment]
What I forgot to check is whether doxygen still works.
Cmake 3.9 (we target 3.7) got additional functionality for woking with doxygen but I don't think they will be very useful for us. https://cmake.org/cmake/help/v3.9/module/FindDoxygen.html

**Pull request merged**